core/lib/zksync_core/src/commitment_generator/mod.rs

```diff
@@ -0,0 +1,235 @@
+    ) -> anyhow::Result<Option<AuxCommitments>> {
+        let mut started_at = Instant::now();
+        let events_queue = conn
+            .blocks_dal()
+            .get_events_queue(l1_batch_number)
+            .await?
+            .context("Events queue is required for post-boojum batch")?;
+        let events_queue_commitment = events_queue_commitment(&events_queue, protocol_version)
+            .context("Events queue commitment is required for post-boojum batch")?;
+        METRICS
+            .events_queue_commitment_latency
+            .observe(started_at.elapsed());
+
+        started_at = Instant::now();
+        let initial_bootloader_contents = conn
+            .blocks_dal()
+            .get_initial_bootloader_heap(l1_batch_number)
+            .await?
+            .context("Bootloader initial heap is missing")?;
+        let bootloader_initial_content_commitment =
+            bootloader_initial_content_commitment(&initial_bootloader_contents, protocol_version)
+                .context("Bootloader content commitment is required for post-boojum batch")?;
+        METRICS
+            .bootloader_content_commitment_latency
+            .observe(started_at.elapsed());
+
+        Ok(Some(AuxCommitments {
+            events_queue_commitment,
+            bootloader_initial_content_commitment,
+        }))
+    }
+
+    async fn prepare_input(
+        connection: &mut StorageProcessor<'_>,
+        l1_batch_number: L1BatchNumber,
+    ) -> anyhow::Result<CommitmentInput> {
+        let header = connection
+            .blocks_dal()
+            .get_l1_batch_header(l1_batch_number)
+            .await?
+            .context("header is missing for batch")?;
+        let tree_data = connection
+            .blocks_dal()
+            .get_l1_batch_tree_data(l1_batch_number)
+            .await?
+            .context("tree data is missing for batch")?;
+
+        // TODO(PLA-731): ensure that the protocol version is always available.
+        let protocol_version = header
+            .protocol_version
+            .unwrap_or_else(ProtocolVersionId::last_potentially_undefined);
+        let common = CommitmentCommonInput {
+            l2_to_l1_logs: header.l2_to_l1_logs,
+            rollup_last_leaf_index: tree_data.rollup_last_leaf_index,
+            rollup_root_hash: tree_data.hash,
+            bootloader_code_hash: header.base_system_contracts_hashes.bootloader,
+            default_aa_code_hash: header.base_system_contracts_hashes.default_aa,
+            protocol_version,
+        };
+        let touched_slots = connection
+            .storage_logs_dal()
+            .get_touched_slots_for_l1_batch(l1_batch_number)
+            .await?;
+        let touched_hashed_keys: Vec<_> =
+            touched_slots.keys().map(|key| key.hashed_key()).collect();
+        let previous_values = connection
+            .storage_logs_dal()
+            .get_previous_storage_values(&touched_hashed_keys, l1_batch_number)
```

Comment:
Dumb question: In metadata calculator, `get_previous_storage_values()` is intentionally avoided (it's sort of an expensive query) in favor of using protective reads to remove "not really touched" slots. Would the same approach work here? Or is it too complex relative to potential performance benefits?